### PR TITLE
Graph version

### PR DESCRIPTION
**Pull request by swilly22**
This PR will continue to fail as long as the new version of RedisGraph-py isn't deployed.
We introduce a `Graph Version` which "signs" the graph schema: (labels, relationship-types and attribute-set)
on each change to the graph "schema" the version is updated, client libraries taking advantage of the compact result-set format need to be in sync with the graph schema, they should update their internal mapping between labels, relationship-types and attribute-set strings to numeric IDs upon graph version change.

### CODE REVIEW

**Review by jeffreylovitz — COMMENTED** [review]
This seems like a sensible change to me. When I test locally with the `graph-version` redisgraph-py branch, I get 8 errors on queries that should have been cached executions reporting that they were uncached. Is that just me?

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]
This is looking good to me, but it definitely merits documentation updates.

File: src/commands/cmd_query.c [hunk 1]
```diff
@@ -72,6 +72,7 @@ void Query_SetTimeOut(uint timeout, ExecutionPlan *plan) {
 	Cron_AddTask(timeout, QueryTimedOut, plan);
 }
 
+
```
[the remark this answers is not in the document]
Comment (swilly22) [reply src/commands/cmd_query.c:75]: 👍

File: src/commands/cmd_dispatcher.c [hunk 2]
```diff
@@ -37,6 +42,24 @@ static int _read_flags(RedisModuleString **argv, int argc, bool *compact, long l
 			continue;
 		}
 
+		if(!strcasecmp(arg, "--version")) {
+			int err = REDISMODULE_ERR;
+			if(i < argc - 1) {
+				i++; // Set the current argument to the timeout value.
```
[the remark this answers is not in the document]
Comment (swilly22) [reply src/commands/cmd_dispatcher.c:48]: 👍

File: src/commands/cmd_dispatcher.c [hunk 3]
```diff
@@ -37,6 +42,24 @@ static int _read_flags(RedisModuleString **argv, int argc, bool *compact, long l
...
+				long long v = GRAPH_VERSION_MISSING;
+				err = RedisModule_StringToLongLong(argv[i], &v);
+				*graph_version = v;
+			}
+
+			// Emit error on missing, negative, or non-numeric timeout values.
```
[the remark this answers is not in the document]
Comment (swilly22) [reply src/commands/cmd_dispatcher.c:54]: 👍

File: src/commands/cmd_dispatcher.c [hunk 4]
```diff
@@ -37,6 +42,24 @@ static int _read_flags(RedisModuleString **argv, int argc, bool *compact, long l
...
+			if(err != REDISMODULE_OK || *graph_version < 0) {
```
[the remark this answers is not in the document]
Comment (swilly22) [reply src/commands/cmd_dispatcher.c:55]: 👍

File: src/procedures/proc_bfs.c [hunk 5]
```diff
@@ -175,6 +175,7 @@ static SIValue *Proc_BFS_Step(ProcedureCtx *ctx) {
 
 	UNUSED(res);
 	res = GxB_MatrixTupleIter_new(&iter, (GrB_Matrix)bfs_ctx->nodes);
+	UNUSED(res);
```
[the remark this answers is not in the document]
Comment (swilly22) [reply src/procedures/proc_bfs.c:178]: 👍

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**